src/charm.py

```diff
@@ -151,6 +159,38 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
             event: a :class:`ConfigChangedEvent` to signal that something happened
         """
         self._configure()
+        if self.unit.is_leader():
+            self._configure_replication(True)
```

Comment:
Should we observe `leader_settings_changed`? If not, should this be:
```suggestion
        self._configure_replication(self.unit.is_leader())
```

Not sure I fully understand the interaction with `_on_peer_data_changed`. Add comment?